specifications/api/swagger.yaml

```diff
@@ -569,17 +569,33 @@ x-transactionMetadata: &transactionMetadata
 
     1. All top-level keys must be integers between `0` and `2^64 - 1`.
 
-    2. Strings must be at most 64 characters
+    2. Each metadata value is tagged with its type.
 
-    3. Bytestrings may be encoded as hex-encoded strings prefix with `0x`
+    3. Strings must be at most 64 bytes when UTF-8 encoded.
 
-    Metadata aren't stored as JSON on the Cardano blockchain but stored using
-    another structured encoding called CBOR which is better suited for binary
-    encoding. JSON can be fully expressed in CBOR but the reciprocal isn't
-    true. So, in order to leverage some of the CBOR optimization, we rely on
-    conventions from JSON. In particular, strings starting with `0x` and that
-    are otherwise encoded as base16 will be decoded and stored as plain byte
-    strings on-chain (saving space and therefore, lowering fees).
+    4. Bytestrings are hex-encoded, with a maximum length of 64 bytes.
+
+    Metadata aren't stored as JSON on the Cardano blockchain but are
+    instead stored using a compact binary encoding (CBOR).
+
+    The binary encoding of metadata values supports three simple types:
+
+    * Integers in the range `-2^32` to `2^32 - 1`.
```

Comment:
It's 64bit integers. -(2^64 - 1) .. 2^64 - 1

Reply:
Oops - thanks @dcoutts, sorry for stirring up confusion but...

... just one thing about this range - I know it's simply an artificial limit and nothing to do with machine representation of signed ints. But there are 2<sup>65</sup> - 1 possible values in this range, which might be unsettling for nerds.

Also there is a function `validateTxMetadataValue` in cardano-api which enforces the bounds [-2<sup>64</sup>, 2<sup>64</sup>]. So  2<sup>65</sup> + 1 possible values. I couldn't find the validation code in the consensus layer or specs.